### ISSUE TITLE
Preprocessor area_statistics broken when using fx_files

### ISSUE THREAD

**Issue by bouweandela** [ESMValGroup/ESMValCore#367]
Example recipe:
```yaml
datasets:
  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 2001, end_year: 2004}

preprocessors:
  global_total:
    area_statistics:
      operator: sum
      fx_files: [areacello]

diagnostics:
  test:
    scripts: null
    variables:
      fgco2:
        preprocessor: global_total
        mip: Omon
 ```

Stack trace:
```
Traceback (most recent call last):
  File "/home/bandela/conda/envs/esmvaltool/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/home/bandela/src/esmvalcore/esmvalcore/_task.py", line 686, in _run_task
    output_files = task.run()
  File "/home/bandela/src/esmvalcore/esmvalcore/_task.py", line 238, in run
    self.output_files = self._run(input_files)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 403, in _run
    product.apply(step, self.debug)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 271, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 216, in preprocess
    result.append(_run_preproc_function(function, item, settings))
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 199, in _run_preproc_function
    return function(items, **kwargs)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/_area.py", line 212, in area_statistics
    grid_areas = tile_grid_areas(cube, fx_files)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/_area.py", line 152, in tile_grid_areas
    grid_areas = da.tile(grid_areas, [cube.shape[0], 1, 1])
  File "/home/bandela/conda/envs/esmvaltool/lib/python3.7/site-packages/dask/array/creation.py", line 800, in tile
    raise NotImplementedError("Only integer valued `reps` supported.")
NotImplementedError: Only integer valued `reps` supported.
"""
```

**Comment by ledm** [comment]
From memory, this issue was caused when someone (not me) switched a numpy.array.tile into dask.tile without realising that the interface was massively different between these two functions with the same name!  

Git blame says that it was @bouweandela. (Sorry Bouwe!)
https://github.com/ESMValGroup/ESMValCore/blame/8f36239120a275193863a9f5f5b9ab79a0c981c3/esmvalcore/preprocessor/_area.py#L147

Although I reviewed that PR too, so Mea culpa!

I believe that this is one of the things where I wrote a fix, but either didn't manage to PR it, or the PR never got reviewed.

Here's where the work left off:
https://github.com/ESMValGroup/ESMValCore/compare/development_fx_area_and_volume

I won't have any time to look at it for a while though - AR6 is killing us right now. All I can say is that there is a solution to this bug in that branch somewhere.

**Comment by bouweandela** [comment]
No worries, I had a go at improving `da.tile` and my PR got merged, so the problem should be solved with the next release of dask. It would have been really helpful if we had unit tests for the `area_statistics` function though, because then we would have noticed before merging the pull request that broke it.

**Comment by bouweandela** [comment]
This issue is solved since dask 2.8
